### DISCUSSION TITLE
tables, plots, media oh my!

### DISCUSSION THREAD

**Discussion by evbo** [raquo/Laminar#121] [Ideas]
[Laminar's web components](https://github.com/uosis/laminar-web-components) don't have a "Discussion" section so I am posting here if that's okay:

tailwind css pretty much covers any range of UI design needs, but what about more complex designs like sortable/aggregatable tables, interactive plots, and rich text / media?

These sorts of high horse power components I'd still rely on React for, which leans me back in that direction for my next project. What are people using for these sorts of components? Could web components grow to include material UI table maybe? Plots?

**Comment by raquo** [comment]
I think the laminar-web-components project you mentioned is more or less abandoned, sadly.

I know people are using https://github.com/sherpal/LaminarSAPUI5Bindings/ and it seems to have some kind of table component so maybe that will work for you. If you look at the source code of LaminarSAPUI5Bindings you'll see that creating Laminar interfaces to existing web components is pretty straightforward, so if you find any suitable table web-component on the web, you can probably easily bring it into your Laminar project.

The same goes for plotting libraries – for those ones it doesn't actually matter much if they're made with web-components or with plain JS (since they usually have custom rendering anyway), you just need to find the library that works and write a small interface to use it from Laminar. ScalablyTyped might help with defining the basic types.
